CMakeLists.txt

```diff
@@ -290,23 +290,30 @@ ENDIF()
 # Enable Third Party Libraries
 # ==================================================================
 
-IF (TPL_ENABLE_BLAS)
-  SET(KOKKOSKERNELS_ENABLE_TPL_BLAS ${TPL_ENABLE_BLAS})
+# user overriding kokkoskernels
+IF (NOT DEFINED KOKKOSKERNELS_ENABLE_TPL_BLAS)
+  IF (TPL_ENABLE_BLAS)
+    SET(KOKKOSKERNELS_ENABLE_TPL_BLAS ${TPL_ENABLE_BLAS})
+  ENDIF()
 ENDIF()
-IF (TPL_ENABLE_MKL)
-  SET(KOKKOSKERNELS_ENABLE_TPL_MKL ${TPL_ENABLE_MKL})
+
+IF (NOT DEFINED KOKKOSKERNELS_ENABLE_TPL_MKL)
+  IF (TPL_ENABLE_MKL)
+    SET(KOKKOSKERNELS_ENABLE_TPL_MKL ${TPL_ENABLE_MKL})
+  ENDIF()
 ENDIF()
 
 IF(${Kokkos_ENABLE_Cuda})
```

Comment:
this is tested with trilinos.

Reply:
@ndellingwood TriBITS _should_ automatically try to discover and enable the BLAS TPL if the user doesn't explicitly turn it on.  Is that no longer the case?

Reply:
That default behavior remains the same.

Reply:
I requested this change by @kyungjoo-kim.  The issue is that BLAS is essentially required for building Trilinos, so there is effectively no way to build KokkosKernels in Trilinos and have it use its own implementation of BLAS functions instead of the ones that call the BLAS library.

Reply:
@mhoemmen As far as I know the TriBITS behavior has not changed, my comment above was about an issue in kokkos-kernels where the combo of CUBLAS is enabled but not explicitly setting TPL_ENABLE_BLAS=ON caused compiler errors. Kyungjoo fixed the underlying issue properly in this PR, previously I made a change to the CMakeLists.txt file that worked around the compiler errors but did not allow for users to properly override kokkos-kernel's default TPL settings.